boot/seal.go

```diff
@@ -777,33 +790,45 @@ func runModeBootChains(rbl, bl bootloader.Bootloader, modeenv *Modeenv, cmdlines
 			} else {
 				kernelPath = filepath.Join(runSnapsDir, info.Filename())
 			}
-			runModeBootChain, err := tbl.BootChain(bl, kernelPath)
+			runModeBootChains, err := tbl.BootChains(bl, kernelPath)
 			if err != nil {
 				return err
 			}
 
-			// get asset chains
-			assetChain, kbf, err := buildBootAssets(runModeBootChain, modeenv)
-			if err != nil {
-				return err
+			foundChain := false
+
+			for _, runModeBootChain := range runModeBootChains {
+				// get asset chains
+				assetChain, kbf, err := buildBootAssets(runModeBootChain, modeenv)
+				if err != nil {
+					return err
+				}
+				if assetChain == nil {
+					continue
+				}
+				var kernelRev string
+				if info.SnapRevision().Store() {
+					kernelRev = info.SnapRevision().String()
+				}
+				chains = append(chains, bootChain{
+					BrandID: model.BrandID(),
+					Model:   model.Model(),
+					// TODO: test this
+					Classic:        model.Classic(),
+					Grade:          model.Grade(),
+					ModelSignKeyID: model.SignKeyID(),
+					AssetChain:     assetChain,
+					Kernel:         info.SnapName(),
+					KernelRevision: kernelRev,
+					KernelCmdlines: cmdlines,
+					kernelBootFile: kbf,
+				})
+				foundChain = true
 			}
-			var kernelRev string
-			if info.SnapRevision().Store() {
-				kernelRev = info.SnapRevision().String()
+
+			if !foundChain {
+				return fmt.Errorf("could not find any valid chain for this model")
```

Comment:
Reminder, this is missing a test.

Reply:
I think we are missing a lot of tests for `runModeBootChain`. I will finish that when I am back.

Reply:
Done.